cloudinit/config/cc_set_hostname.py

```diff
@@ -53,7 +53,7 @@
     "distros": [ALL_DISTROS],
     "frequency": frequency,
     "examples": [
-        "perserve_hostname: true",
+        "preserve_hostname: true",
```

Comment:
D'oh, thank you for this catch. Small typo, but big gap in our unittests. I'm sad out unittests aren't strictly validating the individual keys provided in the examples. I think we need to tweak our [examples unittest for this](https://github.com/canonical/cloud-init/blob/main/tests/unittests/config/test_schema.py#L392-L409).

@wschoot if you get a chance to sign our CLA then I can take your commit as is and merge it. https://ubuntu.com/legal/contributors

Ultimately I'd want to fix our unittests here too.
Here's a diff proposal that just highlighted at least 9 other bugs in 9 other documented examples too by making our full schema in unit tests strict to a given module under test.
```diffdiff --git a/tests/unittests/config/test_schema.py b/tests/unittests/config/test_schema.py
index a367cda1c..c31efcc10 100644
--- a/tests/unittests/config/test_schema.py
+++ b/tests/unittests/config/test_schema.py
@@ -406,6 +406,15 @@ class TestCloudConfigExamples:
         """
         schema = get_schema()
         config_load = safe_load(example)
+        # cloud-init-schema is permissive of additionalProperties at the
+        # top-level.
+        # To validate specific schemas against known documented examples
+        # we need to only define the specific module schema and supply
+        # strict=True.
+        # TODO(Drop to pop/update once full schema is strict)
+        schema.pop("allOf")
+        schema.update(schema["$defs"][schema_id])
+        schema["additionalProperties"] = False
         validate_cloudconfig_schema(config_load, schema, strict=True)
 ```

If you'd be willing to sign the CLA and can provide you with the unittest coverage and schema fixes for those 9 errors in this PR.

Reply:
To validate failures locally once this patch is applied on a system with tox installed:
```
tox -e py3 tests/unittests/config/test_schema.py
```